data/transactions/logic/debugger_legacy.go

```diff
@@ -0,0 +1,376 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package logic
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/logging"
+	"github.com/algorand/go-algorand/protocol"
+)
+
+// LegacyDebuggerHook is an interface that supports the first version of TEAL debuggers.
+// It consists of a set of functions called by eval function during TEAL program execution
+type LegacyDebuggerHook interface {
```

Comment:
I don't think the interface containing the methods should be called `*Hook`. I'm onboard with calling the methods themselves hook if others feel strongly, but I think the container of the methods is not, itself, a hook.

Reply:
I agree. I took this to mean we should rename `DebuggerHook` to just `Debugger`, and I did that in 0e197628fcb3cf4bb5d1da1d16c893df0b841246